experiments/add_test.go

```diff
@@ -12,64 +12,106 @@ var addResultMFloat64 MVector[float64]
 
 func BenchmarkAddVector(b *testing.B) {
 	var r vector3.Vector[float64]
-	a := vector3.New(1., 2., 3.)
+	va := vector3.New(1., 2., 3.)
+	vb := vector3.New(2., 3., 4.)
 
 	for n := 0; n < b.N; n++ {
-		r = r.Add(a)
+		r = r.Add(va).MultByVector(va).
+			Add(va.Reciprocal()).MultByVector(va.Reciprocal()).
+			MultByVector(vb.Reciprocal()).Add(vb.Reciprocal())
 	}
 	addResultFloat64 = r
 	runtime.KeepAlive(r)
 }
 
 func BenchmarkAddMutableVector(b *testing.B) {
 	var r MVector[float64]
-	a := MVector[float64]{1., 2., 3.}
+	va := MVector[float64]{1., 2., 3.}
+	vb := MVector[float64]{2., 3., 4.}
 
 	for n := 0; n < b.N; n++ {
-		r = r.Add(a)
+		r = r.Add(va).MultByVector(va).
+			Add(va.Reciprocal()).MultByVector(va.Reciprocal()).
+			MultByVector(vb.Reciprocal()).Add(vb.Reciprocal())
 	}
 	addResultMFloat64 = r
 	runtime.KeepAlive(r)
 }
 
 func BenchmarkAddMutableVectorInPlace(b *testing.B) {
 	var r MVector[float64]
-	a := MVector[float64]{1., 2., 3.}
+	va := MVector[float64]{1., 2., 3.}
+	vb := MVector[float64]{2., 3., 4.}
 	for n := 0; n < b.N; n++ {
-		r.AddInPlace(a)
+		r.AddInPlace(va)
+		r.MultInPlace(va)
+		r.AddInPlace(va.Reciprocal())
+		r.MultInPlace(va.Reciprocal())
+		r.MultInPlace(vb.Reciprocal())
+		r.AddInPlace(vb.Reciprocal())
 	}
 	addResultMFloat64 = r
 	runtime.KeepAlive(r)
 }
 
 func BenchmarkAddMutableVectorInPlaceUsingPointer(b *testing.B) {
 	var r *MVector[float64] = &MVector[float64]{1, 2, 3}
-	a := MVector[float64]{1., 2., 3.}
+	va := MVector[float64]{1., 2., 3.}
+	vb := MVector[float64]{2., 3., 4.}
 
 	for n := 0; n < b.N; n++ {
-		r.AddInPlace(a)
+		r.AddInPlace(va)
+		r.MultInPlace(va)
+		r.AddInPlace(va.Reciprocal())
+		r.MultInPlace(va.Reciprocal())
+		r.MultInPlace(vb.Reciprocal())
+		r.AddInPlace(vb.Reciprocal())
 	}
 
 	addResultMFloat64 = *r
 	runtime.KeepAlive(r)
 }
 
+func BenchmarkAddMutableVectorInPlaceDirect(b *testing.B) {
+	var r MVector[float64]
+	va := MVector[float64]{1., 2., 3.}
+	vb := MVector[float64]{2., 3., 4.}
+	for n := 0; n < b.N; n++ {
+		r.X = ((r.X+va.X)*va.X+1/va.X)/va.X/vb.X + 1/vb.X
+		r.Y = ((r.Y+va.Y)*va.Y+1/va.Y)/va.Y/vb.Y + 1/vb.Y
+	}
```

Comment:
Add the z component to make it a fair benchmark
```suggestion
		r.Y = ((r.Y+va.Y)*va.Y+1/va.Y)/va.Y/vb.Y + 1/vb.Y
		r.Z = ((r.Z+va.Z)*va.Z+1/va.Z)/va.Z/vb.Z + 1/vb.Z
	}
```